update to matching

LeonidGrekhov opened the pull request:
adjusted matching to work without sklearn nearest neighbor

LeonidGrekhov committed:
nnm and data generation

LeonidGrekhov committed:
nnm and data generation

LeonidGrekhov committed:
nnm and data generation

LeonidGrekhov committed:
caliper matching

LeonidGrekhov committed:
caliper matching

LeonidGrekhov committed:
caliper matching

LeonidGrekhov committed:
nnm2 function

does everything the other functions use to do

LeonidGrekhov committed:
nnm2 function

does everything the other functions use to do

LeonidGrekhov committed:
nnm2 function

does everything the other functions use to do

LeonidGrekhov committed:
update

LeonidGrekhov committed:
update

LeonidGrekhov committed:
update

LeonidGrekhov committed:
update structure

LeonidGrekhov committed:
update structure

LeonidGrekhov committed:
update structure

LeonidGrekhov committed:
create git ignore

LeonidGrekhov committed:
create git ignore

LeonidGrekhov committed:
create git ignore

LeonidGrekhov committed:
Merge pull request #1 from LeonidGrekhov/structure

added files for structure

LeonidGrekhov committed:
Merge pull request #1 from LeonidGrekhov/structure

added files for structure

LeonidGrekhov committed:
Merge pull request #1 from LeonidGrekhov/structure

added files for structure

LeonidGrekhov committed:
clean up

LeonidGrekhov committed:
clean up

LeonidGrekhov committed:
clean up

LeonidGrekhov committed:
logger, data dictionary, restructure

LeonidGrekhov committed:
logger, data dictionary, restructure

LeonidGrekhov committed:
logger, data dictionary, restructure

LeonidGrekhov committed:
logger, data dictionary, restructure

LeonidGrekhov committed:
logger, data dictionary, restructure

LeonidGrekhov committed:
logger, data dictionary, restructure

kostabotnar committed:
Run main() from __maim__.py
Format code

kostabotnar committed:
Run main() from __maim__.py
Format code

kostabotnar committed:
Run main() from __maim__.py
Format code

kostabotnar committed:
Run main() from __maim__.py
Format code

kostabotnar committed:
Run main() from __maim__.py
Format code

kostabotnar committed:
Run main() from __maim__.py
Format code

kostabotnar committed:
Improve logging

kostabotnar committed:
Improve logging

kostabotnar committed:
Improve logging

kostabotnar committed:
Improve logging

kostabotnar committed:
Improve logging

kostabotnar committed:
Improve logging

LeonidGrekhov committed:
update

adjusted to not use nearestNeighbors from sklearn

LeonidGrekhov committed:
update

adjusted to not use nearestNeighbors from sklearn

LeonidGrekhov committed:
update

adjusted to not use nearestNeighbors from sklearn

LeonidGrekhov committed:
update

adjusted to not use nearestNeighbors from sklearn

LeonidGrekhov committed:
update

adjusted to not use nearestNeighbors from sklearn

LeonidGrekhov committed:
update

adjusted to not use nearestNeighbors from sklearn

LeonidGrekhov committed:
update, code clean up

LeonidGrekhov committed:
update, code clean up

LeonidGrekhov committed:
update, code clean up

LeonidGrekhov committed:
update, code clean up

LeonidGrekhov committed:
update, code clean up

LeonidGrekhov committed:
update, code clean up

LeonidGrekhov committed:
clean up

LeonidGrekhov committed:
clean up

LeonidGrekhov committed:
clean up

LeonidGrekhov committed:
clean up

LeonidGrekhov committed:
clean up

LeonidGrekhov committed:
clean up

kostabotnar commented on src/__pycache__/__init__.cpython-39.pyc:
this file should not be tracked by git

kostabotnar commented on src/__pycache__/main.cpython-39.pyc:
this file should not be tracked by git

kostabotnar commented on src/__pycache__/methods.cpython-39.pyc:
this file should not be tracked by git

kostabotnar commented on src/datamodel/__pycache__/Column.cpython-39.pyc:
this file should not be tracked by git

kostabotnar commented on src/datamodel/__pycache__/__init__.cpython-39.pyc:
this file should not be tracked by git

kostabotnar commented on src/main.py:
```diff
@@ -6,67 +13,59 @@
     # Create a DataFrame
     data = pd.DataFrame({
-        'Patient ID': range(1, total_patients + 1),
+        'Patient ID': range(0, total_patients  ),
```
if range starts from 0 with step 1, you can set only the number of items
range(0,N,1) the same as range(0,N) the same as range(N)
The last one is preferable form

kostabotnar commented on src/main.py:
```diff
@@ -6,67 +13,59 @@
\ No newline at end of file
+        logging.debug(f"{count}\n")
+        count += 1
+        logging.debug(f"Treated Patient:\n{row[0]},\nMatched Patient(s):\n{row[1]}\n")
```
I usually save such results in a csv file. It is more convenient to look through it in excel than in terminal

Additionally, if you want to iterate with item index, use `enumerate()` function instead of count variable

kostabotnar commented on src/main.py:
```diff
@@ -6,67 +13,59 @@
+            
+            count += 1
+            f.write(f"Treated Patient:\n{row[0]},\nMatched Patient(s):\n{row[1]}\n")
+        f.write(f"Total matched pairs: {count}\n")
```
Instead of count you can write just `len(matched_pairs)`

kostabotnar commented on src/util/__pycache__/FileProvider.cpython-39.pyc:
this file should not be tracked by git

kostabotnar commented on src/util/__pycache__/__init__.cpython-39.pyc:
this file should not be tracked by git

kostabotnar commented on src/util/__pycache__/methods.cpython-39.pyc:
this file should not be tracked by git

kostabotnar commented on src/util/methods.py:
```diff
@@ -0,0 +1,55 @@
+from sklearn.neighbors import NearestNeighbors
+from scipy.spatial.distance import cdist
+
+def nnm2(medical_data: pd.DataFrame, replacement: int, caliper: float, k: int) -> list:
```
if `replacement` parameter can be only 1 or 0, make it boolean

kostabotnar commented on src/util/methods.py:
```diff
@@ -0,0 +1,55 @@
+
+def nnm2(medical_data: pd.DataFrame, replacement: int, caliper: float, k: int) -> list:
+    """
+    Function description
```
No need in this row. This section is only for the description

kostabotnar commented on src/util/methods.py:
```diff
@@ -0,0 +1,55 @@
+    """
+    Function description
+    Accepts dataframe with parameters of replacement, caliper and number of neighbors
+    returns a list of pairs
```
specify pairs of what

kostabotnar commented on src/util/methods.py:
```diff
@@ -0,0 +1,55 @@
+    Accepts dataframe with parameters of replacement, caliper and number of neighbors
+    returns a list of pairs
+    """
+    logger = logging.getLogger('NNM2')
```
make logger global for the file with the same name as the file

kostabotnar commented on src/util/methods.py:
```diff
@@ -0,0 +1,55 @@
+    """
+    logger = logging.getLogger('NNM2')
+    pairs = []
+    logger.debug(f'print medical data: {medical_data}')
```
It is a bad practice of printing data frame. You may print the shape or some specific parameters. Usually the data frame contains thousands of rows and dozens of columns. If you need to see the data, save it to a temp file.

kostabotnar commented on src/util/methods.py:
```diff
@@ -0,0 +1,55 @@
+        print("Nearest Neighbor with replacement")
+
+    if caliper != 0:
+        print("Nearest Neighbor with caliper")
```
if you want to log function parameters values, log them at the beginning of the  function
`logger.debug(f"data size {df.shape}, replacement: {replacement}, caliper: {caliper}")`

kostabotnar commented on src/util/methods.py:
```diff
@@ -0,0 +1,55 @@
+        if match:
+            pairs.append((treated_unit, nearest_neighbors))
+    logger.debug(f'length of control group after execution  {len(control_group)}')   
+    return pairs
```
Let's try to make this code simpler. The general idea of working with data frame is to work with it  as with table or matrix and avoid working with separate rows or values. I would propose the next approach:
1. In control group create a column DIFF with ABS(propensity_score - treated_item.propensity_score).
2. If caliper : filter control group by DIFF<=caliper_value
3. else: Sort control DF by DIFF in ascending order and get K-first rows
4. As a result you'll have a DF with chosen items. Here we can apply any method to peak up matched item. We decided to take the nearest one. So, get items with the minimal DIFF. You may have several items, so peak a random one.
5. If with replacement, then drop all final selected items (we have only one) from the control group
6. if caliper

LeonidGrekhov committed:
update

LeonidGrekhov committed:
update

LeonidGrekhov committed:
update

LeonidGrekhov committed:
update

LeonidGrekhov committed:
update

LeonidGrekhov committed:
update

LeonidGrekhov committed:
git ignore

LeonidGrekhov committed:
git ignore

LeonidGrekhov committed:
git ignore

LeonidGrekhov committed:
git ignore

LeonidGrekhov committed:
git ignore

LeonidGrekhov committed:
git ignore

kostabotnar commented on .gitignore:
```diff
@@ -1,2 +1,9 @@
+src/__pycache__/
+src/datamodel/__pycache__/
+src/util/__pycache__/
+src/__pycache__/main.cpython-39.pyc
```
to ignore all .pyc files in project `*.pyc` would be enough

kostabotnar commented on .gitignore:
```diff
@@ -1,2 +1,9 @@
 data
 build/
+/test.log
+__pycache__/
```
this string is enough to ignore all __pycache__ folders in the project

kostabotnar commented on src/main.py:
```diff
@@ -47,25 +44,12 @@ def main():
-
-    with open(output_file, "w") as f:
-        count = 0
+    output_file = "data\\output.csv"  # Output file
```
`data` folder is mostly for the input data files and sometimes for the intermediate files. All output file should be kept in the `build` folder. 
Try to give a meaningful names for all you files. For example this one can be  `matched_pairs.csv`
If folder, when you are trying to save a file does not exist in the repository (for example `src` always exists but `build` does not), than you need to check it and create the folder, before writing the file. Otherwise, anyone who clone the repo and run the project will get an error `FileNotFoundError: [Errno 2] No such file or directory: 'data\\output.csv'`

kostabotnar commented on src/main.py:
```diff
@@ -47,25 +44,12 @@ def main():
-            count += 1
             f.write(f"Treated Patient:\n{row[0]},\nMatched Patient(s):\n{row[1]}\n")
-        f.write(f"Total matched pairs: {count}\n")
+        f.write(f"Total matched pairs: {len(matched_pairs)}\n")
```
If you want to create a text file, than give it `.txt` extension. If you are creating a`.csv` file, than it should be formatted correctly as a coma separated data.

LeonidGrekhov committed:
create build folder if doesnt exist

added create a folder if it does not exist and write output file matched_pairs.txt to build folder

LeonidGrekhov committed:
create build folder if doesnt exist

added create a folder if it does not exist and write output file matched_pairs.txt to build folder

LeonidGrekhov committed:
create build folder if doesnt exist

added create a folder if it does not exist and write output file matched_pairs.txt to build folder

LeonidGrekhov committed:
create build folder if doesnt exist

added create a folder if it does not exist and write output file matched_pairs.txt to build folder

LeonidGrekhov committed:
create build folder if doesnt exist

added create a folder if it does not exist and write output file matched_pairs.txt to build folder

LeonidGrekhov committed:
create build folder if doesnt exist

added create a folder if it does not exist and write output file matched_pairs.txt to build folder

kostabotnar committed:
Improve matching approach

kostabotnar committed:
Improve matching approach

kostabotnar committed:
Improve matching approach

kostabotnar committed:
Improve matching approach

kostabotnar committed:
Improve matching approach

kostabotnar committed:
Improve matching approach

kostabotnar committed:
Merge remote-tracking branch 'origin/structure' into structure

kostabotnar committed:
Merge remote-tracking branch 'origin/structure' into structure

kostabotnar committed:
Merge remote-tracking branch 'origin/structure' into structure

kostabotnar committed:
Merge remote-tracking branch 'origin/structure' into structure

kostabotnar committed:
Merge remote-tracking branch 'origin/structure' into structure

kostabotnar committed:
Merge remote-tracking branch 'origin/structure' into structure

kostabotnar committed:
Merge branch 'dev' into structure

# Conflicts:
#	src/main.py
#	src/methods.py

kostabotnar committed:
Merge branch 'dev' into structure

# Conflicts:
#	src/main.py
#	src/methods.py

kostabotnar committed:
Merge branch 'dev' into structure

# Conflicts:
#	src/main.py
#	src/methods.py

kostabotnar committed:
Merge branch 'dev' into structure

# Conflicts:
#	src/main.py
#	src/methods.py

kostabotnar committed:
Merge branch 'dev' into structure

# Conflicts:
#	src/main.py
#	src/methods.py

kostabotnar committed:
Merge branch 'dev' into structure

# Conflicts:
#	src/main.py
#	src/methods.py

kostabotnar committed:
Convert series to data frame

kostabotnar committed:
Convert series to data frame

kostabotnar committed:
Convert series to data frame

kostabotnar committed:
Convert series to data frame

kostabotnar committed:
Convert series to data frame

kostabotnar committed:
Convert series to data frame

kostabotnar committed:
Convert series to data frame

kostabotnar committed:
Convert series to data frame

kostabotnar committed:
Convert series to data frame

kostabotnar committed:
Convert series to data frame

kostabotnar committed:
Convert series to data frame

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Update .gitignore

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
remove files

LeonidGrekhov committed:
remove files

LeonidGrekhov committed:
remove files

LeonidGrekhov committed:
remove files

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure

LeonidGrekhov committed:
organize output

LeonidGrekhov committed:
Merge branch 'structure' of https://github.com/LeonidGrekhov/psm-implementation into structure
